src/internal/modules/returns-notifications/lib/controller.js

```diff
@@ -21,15 +22,23 @@ const checkAnswersRoute = '/returns-notifications/check-answers';
  */
 const createGetHandler = async (request, h, formContainer) => {
   const { document } = request.pre;
+  const form = sessionForms.get(request, formContainer.form(request));
   const view = {
     ...request.view,
-    caption: `Licence ${document.document.licenceNumber}`,
+    caption: document && `Licence ${document.document.licenceNumber}`,
     back: checkAnswersRoute,
-    form: formContainer.form(request, document)
+    form
   };
   return h.view('nunjucks/form', view);
 };
 
+const processAction = (request, action) => {
+  const currentState = request.yar.get(SESSION_KEYS.paperFormsFlow);
+  const nextState = reducer.reducer(currentState, action);
```

Comment:
You could get rid of line 36 and 37 could change to:
`const nextState = reducer.reducer(request.pre.state, action);`

Reply:
I tried this and it didn't work unfortunately, I don't think the pre-handler is needed on all routes.

I have made one small tweak though which is to hide the submit button if no returns are selected.